ads/3p.md

```diff
@@ -0,0 +1,99 @@
+
+#### Ad Agency
+*Examples : Essence, Omnicom*
+
+Work with your publisher to ensure that the creatives you develop are AMP compliant. Since all creatives are served into iframes whose size is determined, when the ad is called, ensure that your creative doesn't try to modify the size of the iframe.
+
+Ensure that all assets that are part of the creative are requested using HTTPS.
+Some ad formats are fully not supported at the moment and we recommend testing the creatives in an AMP environment. Some examples are : Rich Media Expandables, Interstitials, Page Level Ads
+
+#### Video Player
+
+*Examples : Brightcove, Ooyala*
+
+A video player that works in regular HTML pages will not work in AMP and therefore a specific tag needs to be created that will allow the AMP runtime to load your player. 
+Brightcove has already created a custom amp-brightcove tag that allows media and ads to be played in AMP pages.
+
+A brightcove player can be invoked by the following:
+```
+<amp-brightcove
+      data-account="906043040001"
+      data-video-id="1401169490001"
+      data-player-id="180a5658-8be8-4f33-8eba-d562ab41b40c"
+      layout="responsive" width="480" height="270">
+  </amp-brightcove>
+```
+For instructions on how to develop an amp tag like brightcove, please see [amp-brightcove.js](https://github.com/ampproject/amphtml/blob/master/extensions/amp-brightcove/0.1/amp-brightcove.js)
+
+The other popular option is for video players to take advantage of the amp-iframe (https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/amp-iframe.md) extension (with post frame or fallback markup) and serve the player within the iframe. 
+
+#### Video Ad Network
+
+*Examples : Tremor, Brightroll*
+
+If you are a video ad network, please work with your publisher to ensure that :
+
+-All video assets are being served over HTTPS
+-The publisher’s video player has AMP support
+
+
+#### Data Management Platform (DMP)
+*Examples : KRUX, Bluekai*
+
+[See how to enhance custom ad configuration](https://github.com/ampproject/amphtml/blob/master/builtins/amp-ad.md#enhance-incoming-ad-configuration)
+
+You can use a similar approach to enrich the ad call by passing in audience segments that you get from the user cookie into the ad call.
+
+#### Viewability Provider
+
+*Examples : MOAT, Integral Ad Science*
+
+Viewability providers typically integrate with publishers via the ad server’s creative wrappers. If that is the case, ensure that the creative wrapper loads all assets over HTTPS.
+
+For e.g. for MOAT, make sure http://js.moatads.com is switched to  https://z.moatads.com
+
+If you instead would like to directly embed viewability JS on the publisher page, use the approach to inject javascript on a remote server. [See how to enhance custom ad configuration](https://github.com/ampproject/amphtml/blob/master/builtins/amp-ad.md#enhance-incoming-ad-configuration)
+
+Alternatively, see appraoch to using the [intersection obeserver pattern](#ad-viewability)
```

Comment:
There is no alternative here. The intersection observer approach is the only way to do it.